fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/visitor/ExpressionColumnFilterConverter.java

```diff
@@ -0,0 +1,109 @@
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.trees.expressions.visitor;
+
+import org.apache.doris.analysis.Expr;
+import org.apache.doris.analysis.LiteralExpr;
+import org.apache.doris.analysis.NullLiteral;
+import org.apache.doris.analysis.SlotRef;
+import org.apache.doris.nereids.trees.expressions.ComparisonPredicate;
+import org.apache.doris.nereids.trees.expressions.EqualTo;
+import org.apache.doris.nereids.trees.expressions.Expression;
+import org.apache.doris.nereids.trees.expressions.GreaterThan;
+import org.apache.doris.nereids.trees.expressions.GreaterThanEqual;
+import org.apache.doris.nereids.trees.expressions.InPredicate;
+import org.apache.doris.nereids.trees.expressions.IsNull;
+import org.apache.doris.nereids.trees.expressions.LessThan;
+import org.apache.doris.nereids.trees.expressions.LessThanEqual;
+import org.apache.doris.nereids.trees.expressions.NullSafeEqual;
+import org.apache.doris.nereids.trees.expressions.Slot;
+import org.apache.doris.nereids.trees.expressions.literal.Literal;
+import org.apache.doris.planner.PartitionColumnFilter;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * expression column filter converter
+ */
+public class ExpressionColumnFilterConverter
+        extends DefaultExpressionVisitor<Expression, Map<String, PartitionColumnFilter>> {
+    public static ExpressionColumnFilterConverter INSTANCE = new ExpressionColumnFilterConverter();
+
+    public static void convert(Expression expr, Map<String, PartitionColumnFilter> context) {
+        expr.accept(INSTANCE, context);
+    }
+
+    @Override
+    public Expression visitComparisonPredicate(ComparisonPredicate predicate,
+            Map<String, PartitionColumnFilter> columnFilterMap) {
+        if (predicate instanceof NullSafeEqual) {
+            return null;
+        }
+        String colName = ((Slot) predicate.left()).getName();
+        PartitionColumnFilter filter = columnFilterMap.getOrDefault(colName, new PartitionColumnFilter());
+        LiteralExpr literal = ((Literal) predicate.right()).toLegacyLiteral();
+        if (predicate instanceof EqualTo) {
+            setFilter(filter, literal, true, literal, true);
+        } else if (predicate instanceof GreaterThan) {
+            setFilter(filter, literal, false, null, false);
+        } else if (predicate instanceof GreaterThanEqual) {
+            setFilter(filter, literal, true, null, false);
+        } else if (predicate instanceof LessThan) {
+            setFilter(filter, null, false, literal, false);
+        } else if (predicate instanceof LessThanEqual) {
+            setFilter(filter, null, false, literal, true);
```

Comment:
u could use some local variables to record setFilter's parameters and call `setFilter` at the end once for easy reading